chalk-solve/src/infer/unify.rs

```diff
@@ -472,14 +478,15 @@ impl<'t, I: Interner> Unifier<'t, I> {
         Ok(())
     }
 
-    fn push_lifetime_eq_subgoal(&mut self, a: Lifetime<I>, b: Lifetime<I>) {
-        let interner = self.interner;
-        let b_outlives_a = GoalData::AddRegionConstraint(b.clone(), a.clone()).intern(interner);
-        self.goals
-            .push(InEnvironment::new(self.environment, b_outlives_a));
-        let a_outlives_b = GoalData::AddRegionConstraint(a, b).intern(interner);
-        self.goals
-            .push(InEnvironment::new(self.environment, a_outlives_b));
+    fn push_lifetime_eq_constraint(&mut self, a: Lifetime<I>, b: Lifetime<I>) {
```

Comment:
This setup is fine, but I think that the idea would be to push a [`LifetimeOutlives`](http://rust-lang.github.io/chalk/chalk_ir/struct.LifetimeOutlives.html) domain goal, and that we would have chalk-solve have a rule that says "to prove `LifetimeOutlives('a, 'b)`, add the region constraints that `'a: 'b`"

Reply:
Hmm, yeah that's actually probably a better way to approach this.